nflow-engine/src/main/java/io/nflow/engine/internal/executor/WorkflowDispatcher.java

```diff
@@ -80,6 +86,8 @@ public void run() {
           sleep(false);
         }
       }
+    } catch (IOException | ReflectiveOperationException e) {
```

Comment:
Hmm, I still don't quite understand: the above code does not work, as that causes compilation error in the run() method which doesn't throw anything. Same if I remove the catch.

This:

```
    } catch (Exception e) {
      logger.error("Fetching workflow definitions failed", e);
    } finally {
      shutdownPool();
      ....
```

works, but can't wrap my head around how your suggestions do?